### PR TITLE
WIP: feat(custom-js): implement POST /processors

### LINKED ISSUE #189: Implement POST /processors

### LINKED ISSUE #192: Validate javascript validity and throw informative exceptions

### PR DESCRIPTION
**Pull request by NivLipetz**
closes #192 
closes #189

### CODE REVIEW

**Comment by syncush** [comment]
when the PR is accepted, integration tests can be added to #212

File: src/database/cassandra-handler/cassandra.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ const schedulerCassandraConnector = require('../../jobs/models/database/cassandr
 const reportsCassandraConnector = require('../../reports/models/database/cassandra/cassandraConnector');
 const testsCassandraConnector = require('../../tests/models/database/cassandra/cassandraConnector');
 const configCassandraConnector = require('../../configManager/models/database/cassandra/cassandraConnector');
+const processorsSequlizeConnector = require('../../processors/models/database/cassandra/cassandraConnector');
```
Comment (syncush) [inline src/database/cassandra-handler/cassandra.js]: I think the variable name is incorrect

File: src/tests/models/fileManager.js [hunk 2]
```diff
@@ -6,6 +6,7 @@ const database = require('./database'),
 
 module.exports = {
     createFileFromUrl,
+    downloadFile,
```
Comment (enudler) [inline src/tests/models/fileManager.js]: maybe change
        const errMsg = 'Error to read file, throw exception: ' + err;
to
        const errMsg = 'Error to download file: ' + err;

File: src/processors/controllers/processorController.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+'use strict';
+let processorManager = require('../models/processorsManager');
+
+module.exports.createProcessor = function (req, res, next) {
+    return processorManager.createProcessor(req.body)
+        .then(function (result) {
+            return res.status(201).json(result);
```
Comment (enudler) [inline src/processors/controllers/processorController.js]: the response missing created_at/updated_at

**Comment by NivLipetz** [comment]
> I would add validation that the javascript downloaded is 'legit'

sounds good, and this addition will close issue #192

File: src/tests/models/fileManager.js [hunk 4]
```diff
@@ -49,3 +53,30 @@ async function saveFile(fileUrl) {
...
+    let error, errorMessage;
+    try {
+        const syntax = esprima.parseScript(javascriptFileContent, { tolerant: true });
+        const errors = syntax.errors;
+        if (errors.length > 0) {
+            let errorsString = '';
+            for (let i = 0; i < errors.length; i++) {
+                errorsString += errors[i].description + ', ';
+            }
+            errorsString = errorsString.substring(0, errorsString.length - 2);
```
Comment (syncush) [inline src/tests/models/fileManager.js]: I am a fan of oneliners,
this loop can also be written like this
`errorsString = errors.map(error => error.description).join(', ')`

File: src/processors/controllers/processorController.js (hunk 3 above)
Comment (NivLipetz) [reply src/processors/controllers/processorController.js]: done

File: src/tests/models/fileManager.js (hunk 4 above)
Comment (NivLipetz) [reply src/tests/models/fileManager.js]: ended up removing the errors array all-together

**Pull request merged**